### ISSUE TITLE
can Countable support relation or condition ?

### ISSUE THREAD

**Issue by gabrieltong** [kirkbushell/eloquence#51]
Countable is easy to use , but It doesn't work in my case . 

```
post has two state : active or inactive
user can has three relation to post : all post , active posts and inactive posts, 
and I want to cache all three count
```

thanks

**Comment by gabrieltong** [comment]
I checked the code, it's not based on eloquent relation, but db sql, why do we use relation ? so there's no condition problem, I can put the condition in relation .

**Comment by kirkbushell** [comment]
The reason is because the SQL statement helps to avoid race condition issues where 2 requests may be executed at the same time, resulting in an incorrect count on the related table. If you're fetching an object, and then making a change and doing the update that way, you will get incorrect counts over time.

That said, I think we could also do an update() call against the relation, that could work. I think this deserves some more thought, thanks for the heads up!

**Comment by divdax** [comment]
I also need to apply a condition for counting related rows. e.g. User (posts_count) - Post where active = 1.

**Comment by jaulz** [comment]
@kirkbushell is this still somewhere on the roadmap? would you accept a PR?

**Comment by kirkbushell** [comment]
@jaulz I would.

**Comment by kirkbushell** [comment]
@gabrieltong this is solved in #107 - if you have time/moment, would be great to test it out and just double-check. But now, Eloquence support model relationships by default and works with them to drive the behaviours of the library.

**Comment by kirkbushell** [comment]
Closing as this will be available in the next big release for Eloquence.
